### ISSUE TITLE
Crash on 2.5.0

### ISSUE THREAD

**Issue by jakoss** [pandulapeter/beagle#80]
Crash on application run:
```
/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.ku.mobile.android.dev, PID: 29126
    java.lang.NoClassDefFoundError: Failed resolution of: Landroidx/databinding/DataBinderMapperImpl;
        at androidx.databinding.DataBindingUtil.<clinit>(DataBindingUtil.java:32)
        at androidx.databinding.DataBindingUtil.getDefaultComponent(DataBindingUtil.java:65)
        at com.pandulapeter.beagle.core.databinding.BeagleCellHeaderBinding.inflate(BeagleCellHeaderBinding.java:37)
        at com.pandulapeter.beagle.core.list.cells.HeaderCell$createViewHolderDelegate$1.createViewHolder(HeaderCell.kt:22)
        at com.pandulapeter.beagle.core.list.cells.HeaderCell$createViewHolderDelegate$1.createViewHolder(HeaderCell.kt:19)
        at com.pandulapeter.beagle.core.list.CellAdapter.onCreateViewHolder(CellAdapter.kt:36)
        at com.pandulapeter.beagle.core.list.CellAdapter.onCreateViewHolder(CellAdapter.kt:16)
        at androidx.recyclerview.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:7301)
        at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6426)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6310)
        at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6306)
        at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2330)
        at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1631)
        at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1591)
        at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:668)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4309)
        at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4012)
        at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4578)
        at android.view.View.layout(View.java:22844)
        at android.view.ViewGroup.layout(ViewGroup.java:6389)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
        at android.view.View.layout(View.java:22844)
        at android.view.ViewGroup.layout(ViewGroup.java:6389)
        at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1255)
        at android.view.View.layout(View.java:22844)
        at android.view.ViewGroup.layout(ViewGroup.java:6389)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
        at android.view.View.layout(View.java:22844)
        at android.view.ViewGroup.layout(ViewGroup.java:6389)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
        at android.view.View.layout(View.java:22844)
        at android.view.ViewGroup.layout(ViewGroup.java:6389)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
        at android.view.View.layout(View.java:22844)
        at android.view.ViewGroup.layout(ViewGroup.java:6389)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
        at android.view.View.layout(View.java:22844)
        at android.view.ViewGroup.layout(ViewGroup.java:6389)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
        at com.android.internal.policy.DecorView.onLayout(DecorView.java:784)
        at android.view.View.layout(View.java:22844)
        at android.view.ViewGroup.layout(ViewGroup.java:6389)
        at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3470)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2938)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1952)
E/AndroidRuntime:     at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8171)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:972)
        at android.view.Choreographer.doCallbacks(Choreographer.java:796)
        at android.view.Choreographer.doFrame(Choreographer.java:731)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:957)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "androidx.databinding.DataBinderMapperImpl" on path: DexPathList[[zip file "/data/app/~~htSZLh1IavP700g-QPMFbA==/com.ku.mobile.android.dev-t-5BZPrS1SzkKP9Zfosw6w==/base.apk"],nativeLibraryDirectories=[/data/app/~~htSZLh1IavP700g-QPMFbA==/com.ku.mobile.android.dev-t-5BZPrS1SzkKP9Zfosw6w==/lib/x86, /system/lib, /system_ext/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:207)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        	... 65 more
```

I have a special style
```
    <style name="AppTheme.Beagle" parent="Theme.MaterialComponents.Light.NoActionBar">
    </style>
```

And i'm setting it up like this:
```
Beagle.initialize(
            this,
            appearance = Appearance(
                themeResourceId = R.style.AppTheme_Beagle
            ),
            behavior = behavior
        )
```

To me the error seems to be related to `ViewBinding`, but i'm not sure why that could happen

**Comment by egorikftp** [comment]
The same issue.  Looks like Beagle uses DataBinding feature and application can't find necessary dependencies runtime

**Comment by pandulapeter** [comment]
Sorry for messing this up :| 

Indeed, I migrated to data binding internally, to start simplifying the codebase a bit. I wasn't aware that this will affect client applications (and didn't catch the issue as the projects I'm testing with all use data binding). Lesson learned.

I'll revert these changes in the next release, as soon as possible. For the time being, looks like you can solve the issue  by reverting to the previous Beagle version (2.4.4), or enabling data binding in your app module's `build.gradle` (within the `android` block: `buildFeatures.dataBinding = true` )

**Comment by pandulapeter** [comment]
Fixed in [release 2.5.1](https://github.com/pandulapeter/beagle/releases/tag/2.5.1). Now I'm using ViewBinding which seems to work on a per-module basis: consumer apps don't have to enable it for Beagle to work.

Thanks for reporting the issue and for keeping the library up to date on your side, I really appreciate it. New releases often fix minor issues, so it's great to see that you give them a try. I'll try to be more careful in the future.

**Comment by jakoss** [comment]
Happy to test this out. Since we're using Beagle only in internal QA builds there's no risk to break things here
